src/SpinnerKernel.java

```diff
@@ -0,0 +1,168 @@
+import components.map.Map;
+import components.map.Map2;
+
+public class SpinnerKernel<K, V> extends Map2<K, V> {
+    //NOTES: Within the sphere of Map, is there a way
+    //to hard-code the data type of the value to make sure that it
+    //is an Integer? Or is there an alternative way to keep count of the
+    //"size of each slice"?
+
+    //Also, @Override makes it so the compiler picks the correct method, right?
+
+    //build this on top of Map
+    //Is this an ok way to approach building this component, or is there a
+    //better way?
+
+    //Should I just do the NoteCard component instead?
+
+    //I think this would be similar to a linked array, where one holds the
+    //values of the user and the other holds a counter of it.
+
+    //I don't know how effecient this will be oops.
+
+    //these should not be within kernel but in secondary I think.
+    //I need to study up my chart of extends and implements.
+
+    /*
+     * Kernel methods ---------------------------------------------------------
+     */
+
+    //instance that the slice value has not been seen yet, given a value
+    @Override
+    public final void addSlice(K key, V value) {
+        assert key != null : "Violation of: key is not null";
+        assert value != null : "Violation of: value is not null";
+        assert !this.hasKey(key) : "Violation of: key is not in DOMAIN(this)";
+
+        this.Spinner.add(key, value);
+    }
+
+    //instance that the slice value has not seen, given a value
+    @Override
+    public final void addSlice(K key, V value) {
```

Comment:
If this is meant to be an overload of the method above, it has to have different parameters.